play/dbt-materialize/dbt/adapters/materialize/connections.py

```diff
@@ -0,0 +1,62 @@
+# Copyright Materialize, Inc. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+from dbt.adapters.postgres import PostgresConnectionManager
+from dbt.adapters.postgres import PostgresCredentials
+
+import dbt.exceptions
+from dataclasses import dataclass
+from dbt import flags
+from dbt.logger import GLOBAL_LOGGER as logger
+
+
+@dataclass
+class MaterializeCredentials(PostgresCredentials):
+    @property
+    def type(self):
+        return "materialize"
+
+
+class MaterializeConnectionManager(PostgresConnectionManager):
+    TYPE = "materialize"
+
+    @classmethod
+    def open(cls, connection):
+        connection = super().open(connection)
+        # Prevents psycopg connection from automatically opening transactions
+        # More info: https://www.psycopg.org/docs/usage.html#transactions-control
+        connection.handle.autocommit = True
+        return connection
+
+    def commit(self):
+        connection = self.get_thread_connection()
+        if flags.STRICT_MODE:
+            if not isinstance(connection, Connection):
+                raise dbt.exceptions.CompilerException(
+                    f"In commit, got {connection} - not a Connection!"
+                )
+
+        # Instead of throwing an error, quietly log if something tries to commit
+        # without an open transaction.
+        # This is needed because the dbt-adapter-tests commit after executing SQL,
+        # but Materialize can't handle all of the required transactions.
+        # https://github.com/fishtown-analytics/dbt/blob/42a85ac39f34b058678fd0c03ff8e8d2835d2808/test/integration/base.py#L681
+        if connection.transaction_open is False:
```

Comment:
You can do: `if not connection.transaction_open`. Also, in general, you want to avoid using `is` for comparison in Python, as it literally tests whether or not the objects are identical.

The one exception being `if x is None` or `if x is not None`. Otherwise, the guidance is to avoid `is` tests.

Reply:
Thanks, I forgot about this!